### PR TITLE
fix(generate): surface clear errors when glob patterns match files outside project root

### LINKED ISSUE #359: bug: generate ignores files with wildcards specified in config file
There seems to be a bug/regression where it doesn't resolve wildcards in files within config files:

`.cem.yaml`:

```yaml
generate:
  files:
    - '../../packages/components/**/*.ts'
  exclude:
    - '../../packages/components/**/*.spec.ts'
    - '../../packages/components/**/*.stories.ts'
  output: 'custom-elements.json'
```

But when i run `cem generate -v`:

```bash
 DEBUG  Project directory: "/Users/jeroen.zwartepoorte/Projects/components/docs/website"
 DEBUG  Config file: "/Users/jeroen.zwartepoorte/Projects/components/docs/website/.cem.yaml"
Error: pass at least one file to generate
Usage:
  cem generate [files or glob patterns] [flags]
etc.
```

If i add an explicit file to the config, the error is gone.

### PR DESCRIPTION
**Pull request by bennypowers**
## Summary

- `Glob()` now returns sentinel errors for three distinct cases when glob patterns reference files outside the project root, instead of silently returning empty results
- All `Glob()` callers updated to follow the io.Reader pattern: always process returned matches even when error is non-nil
- Fixes the confusing "pass at least one file to generate" error when config file glob patterns legitimately match files but they're all outside the project root

## Context

When a config file specifies glob patterns like `../packages/components/**/*.ts`, `Glob()` correctly expanded them via `doublestar.Glob` but then silently dropped all results because they resolved outside the project root. Users saw only "pass at least one file to generate" with no indication of what went wrong.

The `../` filter is an intentional security boundary and is preserved. This PR surfaces clear errors so users understand what happened:

- `ErrGlobAllOutsideRoot` -- all matched files are outside project root (fatal, exit 1)
- `ErrGlobSomeOutsideRoot` -- some matched files are outside project root (warning, valid results returned)
- `ErrGlobNoneMatched` -- glob pattern matched no files at all (fatal, exit 1)

The return contract follows the `io.Reader` pattern:
- `([], err)` -- fatal, no usable results
- `(files, err)` -- files are valid, err is a warning
- `(files, nil)` -- clean success

## Test plan

- [x] Three new unit tests covering all sentinel error cases
- [x] All 4070 existing tests pass
- [x] Manual test against sl-design-system monorepo confirms clear error message
- [x] `golangci-lint run` clean

Closes #359

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved glob handling across generation, watching, and LSP: glob expansion errors no longer fail the operation when there are valid matches.
  * Demo discovery and negated package patterns now proceed with any returned matches while logging warnings/debug output as needed.
  * Glob results are now more accurate relative to the project root, with clearer outcomes for “none matched” and “matches outside root”.
* **Tests**
  * Added new coverage for glob edge cases, including sentinel error behavior verified with `errors.Is`.

### CODE REVIEW

File: generate/generate.go [hunk 1]
```diff
@@ -81,26 +81,19 @@ func preprocess(ctx types.WorkspaceContext) (r preprocessResult, errs error) {
 		}
 	}
 	if cfg.Generate.DemoDiscovery.FileGlob != "" {
-		demoFiles, err := ctx.Glob(cfg.Generate.DemoDiscovery.FileGlob)
+		demoFiles, _ := ctx.Glob(cfg.Generate.DemoDiscovery.FileGlob)
 		r.demoFiles = demoFiles
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply generate/generate.go]: Demo discovery glob errors now logged at debug level (pterm.Debug). Surfaces hard failures with -v without being noisy.

File: generate/session_watch.go [hunk 2]
```diff
@@ -132,10 +132,7 @@ func (ws *WatchSession) setupWatcher() (*fsnotify.Watcher, error) {
 
 	// Expand globs to get current files and their directories
 	for _, glob := range ws.globs {
-		files, err := ws.ctx.Glob(glob)
-		if err != nil {
-			continue
-		}
+		files, _ := ws.ctx.Glob(glob)
```
[the remark this answers is not in the document]
Comment (bennypowers) [reply generate/session_watch.go:135]: Skipped. Globs already succeeded in expand() before watch mode. Root always watched. Runs every rebuild cycle — logging would repeat.
